### PR TITLE
commands: allow re-editing validation sets on error

### PR DESCRIPTION
**Pull request by sergiusens**
Validation sets on error used to not provide a way to re-edit if the set
 was rejected. Prompt the user if they want to re-edit the set after
 showing the error.

 The change here is the while loop, _generate_template and
 _submit_validation_set have been extracted from the run method.

Signed-off-by: Sergio Schvezov <sergio.schvezov@canonical.com>

- [x] Have you followed the [guidelines for contributing](https://github.com/snapcore/snapcraft/blob/master/CONTRIBUTING.md)?
- [x] Have you signed the [CLA](http://www.ubuntu.com/legal/contributors/)?
- [x] Have you successfully run `make lint`?
- [x] Have you successfully run `pytest tests/unit`?

-----
CRAFT-1388

### CODE REVIEW

File: snapcraft/commands/validation_sets.py [hunk 1]
```diff
@@ -87,50 +88,81 @@ def run(self, parsed_args: "argparse.Namespace"):
...
+
+        while True:
+            try:
+                _submit_validation_set(
+                    edited_validation_sets, parsed_args.key_name, store_client
+                )
+                break
+            except snapcraft_legacy.storeapi.errors.StoreValidationSetsError as validation_error:
+                emit.message(str(validation_error))
+                if not utils.prompt("Do you with to amend the validation set? "):
```
Comment (tigarmo) [inline snapcraft/commands/validation_sets.py:112]:
```suggestion
                if not utils.prompt("Do you wish to amend the validation set? "):
```

File: tests/unit/commands/test_validation_sets.py [hunk 2]
```diff
@@ -229,3 +232,177 @@ def test_edit_validation_sets_with_changes_to_existing_set(
...
+    validation_sets_payload,
+    fake_dashboard_get_validation_sets,
+    fake_dashboard_post_validation_sets_build_assertion,
+    fake_dashboard_post_validation_sets,
+    fake_snap_sign,
+    fake_edit_validation_sets,
+    monkeypatch,
+):
+    fake_edit_validation_sets.return_value = {
+        "account-id": "AccountIDXXXOfTheRequestingUserX",
```
Comment (tigarmo) [inline tests/unit/commands/test_validation_sets.py:248]: This one's just a suggestion: the checks that the tests do are themselves fine, but I think the amount of "hardcoded" text makes it hard to see what's actually being tested. For example, consider putting this dict in a variable that you can use both here and in the `assert fake_dashboard_post_validation_sets_build_assertion.mock_calls` check

**Review by tigarmo — CHANGES_REQUESTED** [review]
Nice, thanks. The test comment is just a suggestion and I'll leave it up to you, but we should fix that typo on the prompt

**Review by mr-cal — APPROVED** [review]
Looks good!

File: tests/unit/commands/test_validation_sets.py (hunk 2 above)
Comment (sergiusens) [reply tests/unit/commands/test_validation_sets.py:248]: sounds like a good suggestion, so yes, I will do this.

**Review by tigarmo — APPROVED** [review]
Leaving this approved to not delay the merge, thanks!

**Pull request merged**